src/components/QuestionTimer.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+
+export default function QuestionTimer({ timeout, onTimeout }) {
+    const [remainingTime, setRemainingTime] = useState(timeout)
+
+    useEffect(() => {
+        setTimeout(onTimeout, timeout)
```

Comment:
To prevent multiple onTimeout triggers, clean up previous timeouts

useEffect(() => {
    const timeoutId = setTimeout(onTimeout, timeout);
    // cleanup
    return () => clearTimeout(timeoutId);  
}, [timeout, onTimeout]);